create.ps1

```diff
@@ -101,278 +89,108 @@ function Get-ErrorMessage {
-                "NoChanges" {
-                    Write-Verbose "No changes needed for account [$($currentAccount.RoepNaam) $($currentAccount.AchterNaam) ($($currentAccount.Id))]"
-
-                    if (-not($dryRun -eq $true)) {
-                        # Set aRef object for use in futher actions
-                        $aRef = [PSCustomObject]@{
-                            Id = $currentAccount.Id
-                        }
-
-                        $auditLogs.Add([PSCustomObject]@{
-                                # Action  = "" # Optional
-                                Message = "No changes needed for account [$($currentAccount.RoepNaam) $($currentAccount.AchterNaam) ($($currentAccount.Id))]"
-                                IsError = $false
-                            })
-                    }
-                    else {
-                        Write-Warning "DryRun: No changes needed for account [$($currentAccount.RoepNaam) $($currentAccount.AchterNaam) ($($currentAccount.Id))]"
-                    }
-
-                    break
-                }
-                #endregion No changes to account
-            }
-
-            # Define ExportData with account fields and correlation property 
-            $exportData = $account.PsObject.Copy() | Select-Object $storeAccountFields
-            # Add correlation property to exportdata
-            $exportData | Add-Member -MemberType NoteProperty -Name $correlation.CorrelationProperty -Value $correlation.CorrelationValue -Force
-            # Add aRef properties to exportdata
-            foreach ($aRefProperty in $aRef.PSObject.Properties) {
-                $exportData | Add-Member -MemberType NoteProperty -Name $aRefProperty.Name -Value $aRefProperty.Value -Force
-            }
+    $currentAccount = $null
+    $currentAccount = Invoke-RestMethod @splatWebRequest -Verbose:$false
+  
+    if ($null -eq $currentAccount) {
+        throw = "Error querying account where [$correlationProperty)] = [$correlationValue]"
+    }
+    Write-Verbose "Correlating to account [$($currentAccount.RoepNaam) $($currentAccount.AchterNaam) ($($currentAccount.Id))]"
 
-            break
+    $outputContext.AccountReference = [PSCustomObject]@{
+        Id = $currentAccount.Id
+    }
+    # Add a message and the result of each of the validations showing what will happen during enforcement
+    if ($actionContext.DryRun -eq $true) {
+        Write-Warning "DryRun: Would correlate to account on field [$correlationProperty] with value: [$($correlationValue)]"
+    }
+    # Process
+    if (-not($actionContext.DryRun -eq $true)) {
+        
+        Write-Verbose 'Correlating user account'
+        $uri = "$($actionContext.Configuration.BaseUri)/medewerkers/$($currentAccount.Id)/$($MutationDate)"
+        $bodyobject = [PSCustomObject]@{}
+        $bodyobject | Add-Member -MemberType NoteProperty -Name EmailZakelijk -Value $actionContext.Data.EmailZakelijk -Force
```

Comment:
I recommend defining an object, such as $account, and mapping it with the data. This approach makes the connector more dynamic, eliminating the need for script changes when adding or modifying field mappings. While not critical, it enhances the connector’s flexibility.